six/modules/c++/six.sicd/source/ComplexXMLParser.cpp

```diff
@@ -1316,6 +1316,22 @@ void ComplexXMLParser::parseImageFormationFromXML(
     ImageFormation *imageFormation) const
 {
     XMLElem tmpElem = getOptional(imageFormationXML, "SegmentIdentifier");
+    XMLElem segmentElem = NULL;
+    XMLElem areaElem = getOptional(getFirstAndOnly(imageFormationXML->getParent(), "RadarCollection"), "Area");
+    if (areaElem) 
+    {
+        XMLElem planeElem = getOptional(areaElem, "Plane");
+        if (planeElem) 
+        {
+            segmentElem = getOptional(planeElem, "SegmentList");
+        }
+    }
+    if (!tmpElem && segmentElem)
```

Comment:
Good catch.  Rather than re-parsing the XML though, you can just check if sicd->radarCollection->area->plane->segmentList exists (with the NULL checks along the way obviously).  I would say pass in a const ref to a `RadarCollection` into this function.  That way it is hopefully obvious up a level that we can't arbitrarily re-order calling this vs. `parseRadarCollectionFromXML()`.